src/components/TagContainer.ts

```diff
@@ -0,0 +1,237 @@
+            : "";
+        const XPath = "//" + this.tagEntity + constraint;
+        mx.data.get({
+            callback: object => {
+                const suggestionList = object.map(tagObject => ({
+                    value: tagObject.get(this.tagAttribute) as string
+                }));
+                this.setState({ suggestions: suggestionList.map(suggestion => suggestion.value) });
+                this.processTags(object);
+            },
+            error: error =>
+                window.mx.ui.error(`An error occurred while retrieving tags via XPath (${this.tagEntity}): ${error}`),
+            xpath: XPath
+        });
+    }
+
+    private processTags(tagObjects: any[]) {
+        const currentTagObjects: mendix.lib.MxObject[] = [];
+        const referenceTags = this.props.mxObject.get(this.referenceAttribute) as string[];
+        tagObjects.map(object => {
+            referenceTags.map(reference => {
+                if (reference === object.getGuid()) {
+                    currentTagObjects.push(object);
+                }
+            });
+        });
+        const tagData = currentTagObjects.map(tagObject => ({ value: tagObject.get(this.tagAttribute) as string }));
+        this.setState({
+            tagList: tagData.map(tag => tag.value)
+        });
+    }
+
+    private createTag(tag: string) {
+        const { afterCreateMicroflow, mxObject } = this.props;
+        mx.data.create({
+            callback: object => {
+                object.set(this.tagAttribute, tag);
+                mx.data.commit({
+                    callback: () => {
+                        mxObject.addReference(this.referenceAttribute, object.getGuid());
+                        this.saveTag(mxObject);
+
+                        if (afterCreateMicroflow) {
+                            this.executeAction(afterCreateMicroflow, object.getGuid());
+                        }
+                    },
+                    error: error => window.mx.ui.error("Error occurred attempting to commit: " + error),
+                    mxobj: object
+                });
+            },
+            entity: this.tagEntity,
+            error: error => window.mx.ui.error(`Error creating tag object ${this.tagEntity}, ${error.message}`)
+        });
+    }
+
+    private saveTag(object: mendix.lib.MxObject) {
+        mx.data.commit({
+            mxobj: object,
+            callback: () => undefined
+        });
+    }
+
+    private removeTag(name: string) {
+        const { onChangeMicroflow, mxObject } = this.props;
+        mx.data.get({
+            callback: (object) => {
+                mxObject.removeReferences(this.referenceAttribute, object[0].getGuid() as any);
+                this.saveTag(mxObject);
+                if (onChangeMicroflow) {
+                    this.executeAction(onChangeMicroflow, object[0].getGuid());
+                }
+            },
+            error: (error) => (`${error.message}, and ${error.stack}`),
```

Comment:
no need for parentheses around `error`

Reply:
you are catching error but all error are handled differently, please make it consistent

Reply:
done